src/test/ui/check-static-recursion-foreign.rs

```diff
@@ -15,7 +15,7 @@ extern crate libc;
 
 use libc::c_int;
 
-#[link_name = "check_static_recursion_foreign_helper"]
+#[link(name = "check_static_recursion_foreign_helper")]
```

Comment:
test break here: https://github.com/rust-lang-ci/rust/runs/984640615#step:23:28391

Reply:
Not sure what's going on here, this definitely should be `#[link]` and not `#[link_name]`.  Maybe it's specific to musl (I'm not too familiar with it)?

Reply:
not sure who to ping about musl: cc @petrochenkov @mati865

Reply:
It's not musl specific, this should fail on (at least some of) other builders as well.
Sorry but I don't have time to investigate it.

Reply:
I'm still a little confused what to do with this.  I agree that removing the attribute entirely is likely fine, but it's unclear to me if this satisfies the intent of the original issue (#18164, I think)

Reply:
It looks like this test was checking for an ICE which would occur before linking and `#[link]` only affects linking so it should be safe to remove this line.